Projects might use a newer CMake version that removes compatibility with CMake 2.8.
This makes builds fail with
>     scikit-build could not get a working generator for your system. Aborting build.

See https://github.com/scikit-build/scikit-build/pull/1159

Author: Alexander Grund (TU Dresden)

--- a/skbuild/platform_specifics/abstract.py
+++ b/skbuild/platform_specifics/abstract.py
@@ -45,7 +45,7 @@ class CMakePlatform(object):
         if not os.path.exists(test_folder):
             os.makedirs(test_folder)
         with open("{:s}/{:s}".format(test_folder, "CMakeLists.txt"), "w") as f:
-            f.write("cmake_minimum_required(VERSION 2.8)\n")
+            f.write("cmake_minimum_required(VERSION ${CMAKE_VERSION})\n")
             f.write("PROJECT(compiler_test NONE)\n")
             for language in languages:
                 f.write("ENABLE_LANGUAGE({:s})\n".format(language))
